### PR TITLE
[netatmo] Consider TimeZoneProvider

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java [hunk 1]
```diff
@@ -39,13 +40,20 @@
 @NonNullByDefault
 public class ChannelTypeUtils {
 
+    private static @Nullable TimeZoneProvider timeZoneProvider;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java:43]: Not sure I'm comfortable with adding static variables if you don't have to. But I can see why you would want to.
@J-N-K What is your opinion on this?

Comment (lolodomo) [reply bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java:43]: I can avoid it but I should then pass TimeZoneProvier as parameter to a lot of handler classes.
If you prefer, I can do it.
Note that the astro binding uses this way with the global variable.

**Comment by lolodomo** [comment]
As you can see, to avoid the global variable, it required many more changes.

Note that I have not tested the changes.

**Comment by cpmeister** [comment]
Yeah, sorry about having you go through that. Static fields can make some things so much more convenient but they can also cause unforeseen problems in what should be a modular environment. Thank you.

**Pull request merged**